File: app/src/main/java/org/michaelbel/moviemade/data/di/component/AppComponent.java
package org.michaelbel.moviemade.data.di.component;

import org.michaelbel.moviemade.data.di.module.ActivityModule;
import org.michaelbel.moviemade.data.di.module.AppModule;
import org.michaelbel.moviemade.data.di.module.FragmentModule;
import org.michaelbel.moviemade.data.di.module.NetworkModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class, NetworkModule.class})
public interface AppComponent {
    ActivityComponent plus(ActivityModule target);
    FragmentComponent plus(FragmentModule target);
}

File: app/src/main/java/org/michaelbel/moviemade/presentation/di/component/AppComponent.kt
package org.michaelbel.moviemade.presentation.di.component

import dagger.Component
import org.michaelbel.moviemade.presentation.di.module.ActivityModule
import org.michaelbel.moviemade.presentation.di.module.AppModule
import org.michaelbel.moviemade.presentation.di.module.FragmentModule
import org.michaelbel.moviemade.presentation.di.module.NetworkModule
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class, NetworkModule::class])
interface AppComponent {
    fun plus(target: ActivityModule): ActivityComponent
    fun plus(target: FragmentModule): FragmentComponent
}